crates/dbsp/src/operator/communication/exchange.rs

```diff
@@ -9,20 +9,43 @@ use crate::{
     circuit::{
         metadata::OperatorLocation,
         operator_traits::{Operator, SinkOperator, SourceOperator},
-        OwnershipPreference, Runtime, Scope,
+        Host, LocalStoreMarker, OwnershipPreference, Runtime, Scope,
     },
     circuit_cache_key,
+    trace::{unaligned_deserialize, Rkyv},
 };
+
 use crossbeam_utils::CachePadded;
-use once_cell::sync::OnceCell;
+use futures::{future, prelude::*};
+use once_cell::sync::{Lazy, OnceCell};
+use rkyv::to_bytes;
 use std::{
     borrow::Cow,
+    collections::HashMap,
     marker::PhantomData,
+    net::SocketAddr,
+    ops::Range,
     sync::{
         atomic::{AtomicUsize, Ordering},
-        Arc, Mutex,
+        Arc, Mutex, RwLock,
     },
 };
+use tarpc::{
+    client, context,
+    serde_transport::tcp::{connect, listen},
+    server::{self, Channel},
+    tokio_serde::formats::Bincode,
+    tokio_util::sync::{CancellationToken, DropGuard},
+};
+use tokio::{
+    runtime::Runtime as TokioRuntime,
+    sync::{Notify, OnceCell as TokioOnceCell},
+    time::sleep,
+};
+use typedmap::TypedMapKey;
+
+// All circuits can share a single Tokio runtime.
+static TOKIO: Lazy<TokioRuntime> = Lazy::new(|| TokioRuntime::new().unwrap());
```

Comment:
I think I have to get a `TokioRuntime` from somewhere, and keep it somewhere, because dropping it shuts down the runtime and a `Handle` does not prevent that. If not by creating it myself, where should I get it from?